### PR TITLE
Create hwintrinsic.cpp

### PR DESCRIPTION
**Pull request by sdmaclea**
Start sharing hwintrinsic code.

@tannergooding @CarolEidt @fiigii PTAL

### CODE REVIEW

File: src/jit/hwintrinsicArm64.cpp [hunk 1]
```diff
@@ -213,6 +169,21 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
...
+        if (simdBaseType == TYP_UNKNOWN)
+        {
+            // TODO-FIXME Add CORINFO_HELP_THROW_TYPE_NOT_SUPPORTED
+            unsigned CORINFO_HELP_THROW_TYPE_NOT_SUPPORTED = CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED;
+
+            return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_TYPE_NOT_SUPPORTED, method, sig, mustExpand);
+        }
+        simdType = getSIMDTypeForSize(simdSizeBytes);
+    }
+
```
Comment (sdmaclea) [inline src/jit/hwintrinsicArm64.cpp]: Merge error...

**Comment by tannergooding** [comment]
Other than the merge conflict, this LGTM.

File: src/jit/hwintrinsicArm64.cpp [hunk 2]
```diff
@@ -213,6 +169,21 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
...
     }
 
+    // Simd instantiation type check
+    if (simdClass != nullptr)
+    {
+        simdBaseType = getBaseTypeAndSizeOfSIMDType(simdClass, &simdSizeBytes);
+
+        if (simdBaseType == TYP_UNKNOWN)
+        {
+            // TODO-FIXME Add CORINFO_HELP_THROW_TYPE_NOT_SUPPORTED
```
Comment (CarolEidt) [inline src/jit/hwintrinsicArm64.cpp]: Nit: Please follow the conventions for `TODO` comments (https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#715-todo) - e.g. I think this should probably be `TODO-Bug`

Comment (sdmaclea) [reply src/jit/hwintrinsicArm64.cpp]: 👍 I'll fix this in #16085 (where this stray bit of code came from)

Comment (CarolEidt) [reply src/jit/hwintrinsicArm64.cpp]: Sounds good. Thanks

**Comment by fiigii** [comment]
Thanks! Please run some hwintrinsic tests before merge.

**Comment by sdmaclea** [comment]
Fixed merge issue.

@tannergooding @fiigii Can you trigger X86 HW intrinsic tests.  I have run on Arm64 already.

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

**Comment by tannergooding** [comment]
test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

**Comment by sdmaclea** [comment]
test Tizen armel Cross Checked Innerloop Build and Test

**Comment by sdmaclea** [comment]
`Ubuntu x64 Checked Build and Test (Jit - EnableIncompleteISAClass=1)`
Failure unrelated to this change and can be ignored.
`
FAILED   - GC/Scenarios/LeakWheel/leakwheel/leakwheel.sh
`

**Comment by sdmaclea** [comment]
Please merge

**Pull request merged**